spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/functions/IcebergVersionFunction.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.spark.functions;
+
+import org.apache.iceberg.IcebergBuild;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.connector.catalog.functions.BoundFunction;
+import org.apache.spark.sql.connector.catalog.functions.ScalarFunction;
+import org.apache.spark.sql.connector.catalog.functions.UnboundFunction;
+import org.apache.spark.sql.types.DataType;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.StructType;
+import org.apache.spark.unsafe.types.UTF8String;
+
+/** Implementation of {@link UnboundFunction} that prints the current Iceberg version in use. */
+public class IcebergVersionFunction implements UnboundFunction {
+  @Override
+  public BoundFunction bind(StructType inputType) {
+    if (inputType.fields().length > 0) {
+      throw new UnsupportedOperationException(
+          String.format(
+              "Invalid input type. Expected 0 fields but found %s", inputType.fields().length));
+    }
+
+    return new IcebergVersion();
+  }
+
+  @Override
+  public String description() {
+    return "iceberg_version - Returns the runtime Iceberg version for debugging purposes";
+  }
+
+  @Override
+  public String name() {
+    return "iceberg_version";
+  }
+
+  public static class IcebergVersion implements ScalarFunction<String> {
+
+    private static final UTF8String icebergBuildVersion =
+        UTF8String.fromString(IcebergBuild.version());
+
+    public static UTF8String invoke() {
+      return icebergBuildVersion;
+    }
+
+    @Override
+    public DataType[] inputTypes() {
+      return new DataType[0];
+    }
+
+    @Override
+    public DataType resultType() {
+      return DataTypes.StringType;
+    }
+
+    @Override
+    public String canonicalName() {
+      return "org.apache.iceberg.spark.functions.iceberg_version()";
+    }
+
+    @Override
+    public String name() {
+      return "iceberg_version";
+    }
+
+    @Override
+    public String produceResult(InternalRow input) {
```

Comment:
Why does this return `String` instead of `UTF8String`?

Reply:
According to Javadoc, the [Spark data type to Java type mapping](https://spark.apache.org/docs/latest/api/java/org/apache/spark/sql/connector/catalog/functions/ScalarFunction.html) applies to the `invoke` functions as well as the values returned by `produceResult`.

Reply:
For the non-magic function, it uses the decoded / deserialized type vs the primitive type or the built-in boxed type (for `UTF8String` etc).

this will actually never call `produceResult`, it should always use the magic function. But I’ve implemented it anyway in case somebody disables magic functions etc.

Reply:
> According to Javadoc, the [Spark data type to Java type mapping](https://spark.apache.org/docs/latest/api/java/org/apache/spark/sql/connector/catalog/functions/ScalarFunction.html) applies to the invoke functions as well as the values returned by produceResult.

Oh I see you're right. Let me look at the plan and see what it's doing. Based on the test implementations I saw, I thought that `String` was used. It would be easier if `UTF8String` is used.

Reply:
Can you also make sure that this is tested?

Reply:
Sure. It is tested in the FunctionCatalog suite, but I'll make it more explicit via refactor.

Reply:
I made the test for it much more explicit, using `Assert.assertEquals(UTF8String.fromString(IcebergBuild.version()), func.produceResult())`. Which fails if it's not UTF8String.